components/director/internal/model/app_template_version.go

```diff
@@ -0,0 +1,41 @@
+package model
+
+import (
+	"encoding/json"
+	"time"
+)
+
+// ApplicationTemplateVersion represents a struct for Application Template Version data
+type ApplicationTemplateVersion struct {
+	ID                    string
+	Version               string
+	Title                 *string
+	ReleaseDate           *string
+	CorrelationIDs        json.RawMessage
+	CreatedAt             time.Time
+	ApplicationTemplateID string
+}
+
+// ApplicationTemplateVersionInput represents a struct with the updatable fields
+type ApplicationTemplateVersionInput struct {
+	Version        string
+	Title          *string
+	ReleaseDate    *string
+	CorrelationIDs json.RawMessage
+}
+
+// ToApplicationTemplateVersion converts ApplicationTemplateVersionInput into ApplicationTemplateVersion
+func (a *ApplicationTemplateVersionInput) ToApplicationTemplateVersion(id, appTemplateID string) ApplicationTemplateVersion {
+	if a == nil {
+		return ApplicationTemplateVersion{}
```

Comment:
Maybe we should return nil instead of empty object?

Reply:
At the current state of the code, I do not see a benefit of that. If we return a nil, we shall need to do additional checks at the places where we use this function.